x/wasm/types/proposal.go

```diff
@@ -852,6 +852,17 @@ func (p PinCodesProposal) ValidateBasic() error {
 	if len(p.CodeIDs) == 0 {
 		return sdkerrors.Wrap(ErrEmpty, "code ids")
 	}
+	for _, num1 := range p.CodeIDs {
+		if num1 == 0 {
+			return sdkerrors.Wrap(ErrZero, "0 is not accepted")
+
+		}
+		for _, num2 := range p.CodeIDs {
+			if num1 == num2 {
```

Comment:
This will always end in a match as you are iterating over the same value set. You need to skip num 1 position.